src/abstractdataframe/abstractdataframe.jl

```diff
@@ -930,7 +940,22 @@ Base.filter((cols, f)::Pair{<:AbstractVector{Symbol}}, df::AbstractDataFrame) =
 Base.filter((cols, f)::Pair, df::AbstractDataFrame) =
     filter(index(df)[cols] => f, df)
 
-_filter_helper(df, f, cols...) = df[((x...) -> f(x...)::Bool).(cols...), :]
+function _filter_helper(df, f, cols...)
```

Comment:
```suggestion
function _filter_helper(df::AbstractDataFrame, f, cols...)
```
(Would any type restriction apply to `cols`?)

Reply:
`AbstractVector` - I will add this.